### PR TITLE
Refactor vmbuilder

### PR DESCRIPTION
**Pull request by futuretea**
Signed-off-by: futuretea <Hang.Yu@suse.com>

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

tests, terraform-provider-harvester, docker-machine-driver-harvester all need to use vmbuilder

**Solution:**

refactor vmbuilder and move it to pkg/builder
tests, terraform-provider-harvester, docker-machine-driver-harvester can import this package

**Related Issue:**
none

**Test plan:**

ci

### CODE REVIEW

**Review by guangbochen — APPROVED** [review]
LGTM

File: pkg/builder/vm.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	}
+	for key, value := range annotations {
+		v.VirtualMachine.ObjectMeta.Annotations[key] = value
+	}
+	return v
+}
+
+func (v *VMBuilder) Memory(memory string) *VMBuilder {
+	v.VirtualMachine.Spec.Template.Spec.Domain.Resources.Requests = corev1.ResourceList{
+		corev1.ResourceMemory: resource.MustParse(memory),
```
Comment (gitlawr) [inline pkg/builder/vm.go:143]: Is it ok if `resource.MustParse` panics with an invalid input?

File: pkg/builder/vm.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+func (v *VMBuilder) VM() *kubevirtv1.VirtualMachine {
+	if v.VirtualMachine.Spec.Template.ObjectMeta.Annotations == nil {
+		v.VirtualMachine.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+	}
+	sshNames, err := json.Marshal(v.SSHNames)
+	if err != nil {
+		return v.VirtualMachine
+	}
```
Comment (gitlawr) [inline pkg/builder/vm.go]: Should we handle the error?

File: pkg/builder/vm.go (hunk 1 above)
Comment (futuretea) [reply pkg/builder/vm.go:143]: yes

**Review by gitlawr — COMMENTED** [review]
LGTM and can be squashed

**Pull request merged**